_posts/2018-12-20-data-modelling-many-to-many.adoc

```diff
@@ -0,0 +1,179 @@
+
+@Temporal(TemporalType.DATE)
+@Column(name = "VALID_UNTIL")
+protected Date validUntil;
+----
+
+
+* https://github.com/cuba-guides/cuba-petclinic-data-model-many-to-many/blob/master/modules/global/src/com/haulmont/sample/petclinic/views.xml[views.xml] - the `pet-with-owner-and-type-and-memberships` view of the pet editing screen contains the composition of `insuranceMemberships` (referencing the `InsuranceMembership` joining entity) with `insuranceCompany` and the additional validity information: `validFrom` and `validUntil`.
+
++
+The `insuranceCompany-with-memberships` view follows the same logic: it includes the composition of `memberships` (referencing the `InsuranceMembership` joining entity) with `pet` and the additional validity information: `validFrom` and `validUntil`.
+
+* https://github.com/cuba-guides/cuba-petclinic-data-model-many-to-many/blob/master/modules/web/src/com/haulmont/sample/petclinic/web/pet/pet/pet-edit.xml[pet-edit.xml] - the XML descriptor of the pet editor defines a datasource for the `Pet` instance and a nested one for its `insuranceCompany` instances (through the `InsuranceMembership` relationship). It also contains a table displaying memberships and the custom action to pick an `InsuranceCompany` directly, bypassing the `InsuranceMembership` editor.
+
+As a result, editing of a `InsuranceMembership` instance works as follows:
+
+The `Pet` edit screen shows a list of airports and the currency drop-down.
+
+
+A user can click *Add Insurance*, the `InsuranceCompany` lookup will be opened, and the user can either select an insurance company to add or create a new insurance company. When the user selects an insurance company, the new `InsuranceMembership` instance https://github.com/cuba-guides/cuba-petclinic-data-model-many-to-many/blob/master/modules/web/src/com/haulmont/sample/petclinic/web/pet/pet/PetEdit.java#L57[is created] with the default validity range. This instance it is not saved to the database directly, but added to the `insuranceMembershipsDs` datasource of the `Pet` editor.
+
+TIP: When within the above described creation of a many-to-many relationship from the Pet editor a `InsuranceCompany` is created or updated through the https://github.com/cuba-guides/cuba-petclinic-data-model-many-to-many/blob/master/modules/web/src/com/haulmont/sample/petclinic/web/insurance/insurancecompany/InsuranceCompanyEdit.java[`InsuranceCompany` editor], the created / updated instance of the insurance company is directly saved both to the database and to the https://github.com/cuba-guides/cuba-petclinic-data-model-many-to-many/blob/master/modules/web/src/com/haulmont/sample/petclinic/web/pet/pet/pet-edit.xml#L11[`insuranceMembershipsDs` datasource] of the Pet editor, as the `InsuranceCompany` entity is fully independent. The user can create new insurance companies and delete existing ones. All changes will be saved to the database in separate transactions and to the `insuranceMembershipsDs` datasource as well.
```

Comment:
I think "is directly saved to the database, as the `InsuranceCompany` entity is fully independent." would be enough